ivy/func_wrapper.py

```diff
@@ -567,6 +569,68 @@ def _wrapped(func):
     return _wrapper_outer
 
 
+# nans Handling #
+# --------------#
+
+
+def _has_nans(x):
+    if isinstance(x, ivy.Container):
+        return x.has_nans()
+    elif isinstance(x, ivy.Array) or ivy.is_native_array(x):
+        return ivy.isnan(x).any()
+    elif isinstance(x, tuple):
+        return any(_has_nans(xi) for xi in x)
+    else:
+        return False
+
+
+def handle_nans(fn: Callable) -> Callable:
+    @functools.wraps(fn)
+    def new_fn(*args, **kwargs):
+        """
+        Checks for the existence of nans in all arrays in the `args`
+        and `kwargs`. The presence of nans is then handled depending
+        on the enabled `nan_policy`.
+
+        Following policies apply:
+        raise_exception: raises an exception in case nans are present
+        warns: warns a user in case nans are present
+        nothing: does nothing
+
+        Parameters
+        ----------
+        args
+            The arguments to be passed to the function.
+        kwargs
+            The keyword arguments to be passed to the function.
+
+        Returns
+        -------
+            The return of the function, with handling of inputs based
+            on the selected `nan_policy`.
+        """
+        # skip the check if the current nan policy is `nothing``
+        if ivy.get_nan_policy() == "nothing":
```

Comment:
we should cache this beforehand:
`nan_policy = ivy.get_nan_policy()`

Reply:
Resolved in https://github.com/unifyai/ivy/pull/6748/commits/130b1ef83c5f6b13681370271c0b9cb767993003